lib/components/BucketTaskCard.dart

```diff
@@ -0,0 +1,326 @@
+import 'package:dotted_border/dotted_border.dart';
+import 'package:provider/provider.dart';
+import 'package:vikunja_app/models/task.dart';
+import 'package:vikunja_app/pages/list/task_edit.dart';
+import 'package:vikunja_app/stores/list_store.dart';
+import 'package:vikunja_app/utils/misc.dart';
+import 'package:vikunja_app/theme/constants.dart';
+
+enum DropLocation {above, below, none}
+
+class TaskData {
+  final Task task;
+  final Size size;
+  TaskData(this.task, this.size);
+}
+
+class BucketTaskCard extends StatefulWidget {
+  final Task task;
+  final int index;
+  final DragUpdateCallback onDragUpdate;
+  final void Function(Task, int) onAccept;
+
+  const BucketTaskCard({
+    Key key,
+    @required this.task,
+    @required this.index,
+    @required this.onDragUpdate,
+    @required this.onAccept,
+  }) : assert(task != null),
+       assert(index != null),
+       assert(onDragUpdate != null),
+       assert(onAccept != null),
+       super(key: key);
+
+  @override
+  State<BucketTaskCard> createState() => _BucketTaskCardState();
+}
+
+class _BucketTaskCardState extends State<BucketTaskCard> with AutomaticKeepAliveClientMixin {
+  Size _cardSize;
+  bool _dragging = false;
+  DropLocation _dropLocation = DropLocation.none;
+  TaskData _dropData;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    if (_cardSize == null) _updateCardSize(context);
+
+    final taskState = Provider.of<ListProvider>(context);
+    final bucket = taskState.buckets[taskState.buckets.indexWhere((b) => b.id == widget.task.bucketId)];
+    // default chip height: 32
+    const double chipHeight = 28;
+    const chipConstraints = BoxConstraints(maxHeight: chipHeight);
+    final theme = Theme.of(context);
+
+    final identifierRow = Row(
+      children: <Widget>[
+        Text(
+          widget.task.identifier.isNotEmpty
+              ? '#${widget.task.identifier.substring(1)}' : '${widget.task.id}',
+          style: theme.textTheme.subtitle2.copyWith(
+            color: Colors.grey,
+          ),
+        ),
+      ],
+    );
+    if (widget.task.done) {
+      identifierRow.children.insert(0, Container(
+        constraints: chipConstraints,
+        padding: EdgeInsets.only(right: 4),
+        child: FittedBox(
+          child: Chip(
+            label: Text('Done'),
+            labelStyle: theme.textTheme.labelLarge.copyWith(
+              fontWeight: FontWeight.bold,
+              color: theme.brightness == Brightness.dark
+                  ? Colors.black : Colors.white,
+            ),
+            backgroundColor: vGreen,
+          ),
+        ),
+      ));
+    }
+
+    final titleRow = Row(
+      children: <Widget>[
+        Expanded(
+          child: Text(
+            widget.task.title,
+            style: theme.textTheme.titleMedium.copyWith(
+              color: widget.task.textColor,
+            ),
+          ),
+        ),
+      ],
+    );
+    if (widget.task.dueDate.year > 2) {
```

Comment:
Why the `> 2`? Ar you checking if there's a due date?

Reply:
Yes, this was a copy from elsewhere in the app (I believe it was the TaskTile?). I wasn't sure how the frontend does this. This check should probably be done in the Task model.

Reply:
Moved the check to `task.dart`, also made it a little more like the frontend in that it does `dueDate.year != 1`.